src/main/kotlin/org/flaxo/plagiarism/model/GraphLink.kt

```diff
@@ -29,5 +29,10 @@ class GraphLink(
          *
          * *The higher weight the closer nodes are.*
          */
-        val weight: Int
+        val weight: Int,
+
+        /**
+         * Direction of the link
+         */
+        val directedTo: Direction
```

Comment:
It is a bit frustrating that in the resulting json file `Direction` enum values are deserialized from strings in uppercase. The problem is that `first` and `second` field names that it refers to are in lowercase.
```
{
    "first": "student1",
    "second": "student2",
    "weight": 75,
    "directedTo": "FIRST"
}
```

Could you please check if overriding `toString` method solves the problem. Otherwise, it can be solved using `kotlinx.serialization` api but probably it is a problem for another issue.

Reply:
I added the serializer, so the names now may be typed in lowercase